spinnaker/api/project.go

```diff
@@ -0,0 +1,105 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/mitchellh/mapstructure"
+	gate "github.com/spinnaker/spin/cmd/gateclient"
+)
+
+func GetProject(client *gate.GatewayClient, projectName string, dest interface{}) error {
+	// todo: find what can be used to get projects
```

Comment:
Still looking around for what can be used here. `spin` currently only has [`AllPipelinesForProjectUsingGET`](https://github.com/spinnaker/spin/blob/master/gateapi/project_controller_api.go#L37) expose from the swagger generated gate-client. Gate has a [`ProjectsController.all()`](https://github.com/spinnaker/gate/blob/master/gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/ProjectController.groovy#L35) endpoint that is used in [`deck`](https://github.com/spinnaker/deck/blob/master/app/scripts/modules/core/src/projects/service/ProjectReader.ts#L8) to enable that UI, but it doesn't seem to be coming through the swagger-spec.

Reply:
@cj-taylor i'll submit a PR to `spin` to add the rest of the Projects endpoints.

Reply:
@cj-taylor those both look good to me. Feel free to PR them. :)

Reply:
@ethanfrogers - opened Issue per the community contribution guidelines to open up communication with all maintainers https://github.com/spinnaker/spinnaker/issues/3787.

PRs opened: 
- https://github.com/spinnaker/gate/pull/690
- https://github.com/spinnaker/spin/pull/104